core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1193,8 +1193,10 @@ class DAGScheduler(
         // TODO: This will be really slow if we keep accumulating shuffle map stages
         for ((shuffleId, stage) <- shuffleToMapStage) {
           stage.removeOutputsOnExecutor(execId)
-          val locs = stage.outputLocs.map(list => if (list.isEmpty) null else list.head).toArray
-          mapOutputTracker.registerMapOutputs(shuffleId, locs, changeEpoch = true)
+          if (!runningStages.contains(stage)) {
```

Comment:
Another things to claim, if we fix the problem described in that issues, we may never meet some problem resulted  by that 2 part code.  it may not help for that issue, do I need to create other issue for that 2.